core/src/main/java/cz/xtf/core/namespace/NamespaceManager.java

```diff
@@ -0,0 +1,191 @@
+package cz.xtf.core.namespace;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.engine.descriptor.MethodBasedTestDescriptor;
+import org.junit.platform.engine.TestDescriptor;
+
+import cz.xtf.core.config.OpenShiftConfig;
+import cz.xtf.core.context.TestCaseContext;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.core.waiting.SimpleWaiter;
+import io.fabric8.kubernetes.api.builder.Visitor;
+import io.fabric8.kubernetes.api.model.NamespaceBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class NamespaceManager {
+    /**
+     * Used only in case when test cases are executed in different namespaces when:
+     * -Dxtf.openshift.namespace.per.testcase=true
+     *
+     * Maps testcase -> namespace
+     */
+    private static final Map<String, String> testcaseToNamespaceMap = new HashMap<String, String>();
+
+    /**
+     * Used only in case when test cases are executed in different namespaces when:
+     * -Dxtf.openshift.namespace.per.testcase=true
+     *
+     * @return Map testcase -> namespace
+     */
+    private static Map<String, String> getTestcaseToNamespaceMap() {
+        return testcaseToNamespaceMap;
+    }
+
+    /**
+     * Used only in case when test cases are executed in different namespaces when:
+     * -Dxtf.openshift.namespace.per.testcase=true
+     *
+     * @return return namespace for currently running testcase or null if not present
+     */
+    public static String getNamespaceForTestcase(String testcase) {
```

Comment:
I see you are using in this MR both `Testcase` and `TestCase` inconsistly. 

My personal attitude is `TestCase` is correct one.